how-it-works/data-and-provider-model/buckets-files-and-merkle-proofs.md

```diff
@@ -0,0 +1,98 @@
+---
+title: Buckets, Files, and Merkle Proofs
+description: High-level overview of how DataHaven organizes data and proves integrity using buckets, files, and Merkle proofs.
+slug: /how-it-works/data-and-provider-model/buckets-files-merkle-proofs
+sidebar_label: Buckets, Files & Merkle Proofs
+---
+
+## Introduction
+
+Storing large files directly on-chain is impractical, while relying on off-chain storage alone forces you to trust whoever holds the bytes. DataHaven resolves this trade-off by separating storage from verification: providers store the data off-chain, and the chain anchors compact cryptographic commitments on-chain.
+
+In practice, DataHaven groups objects into buckets (user-created, provider-hosted containers), commits each file by its Merkle root (a unique fingerprint of the file’s chunks), and uses Merkle proofs to verify integrity at read time and during periodic challenges, without moving whole files on-chain. This page explains those three building blocks and how they fit together so apps and smart contracts can verify existence, integrity, and ongoing availability with minimal on-chain impact. 
+
+## Buckets: Purpose and Structure
+
+A bucket groups related objects under a single provider and policy. Think of it as a scoped namespace with:
+
+- **Ownership & policy:** Who can write/read, retention expectations, redundancy targets.
+- **Compact on-chain summary:** The chain records a cryptographic summary, not every file.
+- **Stable anchor:** One place for pricing, accounting, and proofs across contained files.
+
+Each provider can maintain one or more buckets. Buckets are summarized on-chain so smart contracts can answer questions like "did this object exist at block N?" without bloating state. DataHaven uses Merkle/Patricia-style structures to keep on-chain state minimal yet verifiable.
+
+### Bucket types and organization
+
+Create a new bucket to organize and store your data. Choose **Public** for shared access or **Private (encrypted)** for sensitive information.
+
+- **Public bucket**: Designed for shared or team access. Good for assets meant to be broadly readable by your app or collaborators.
+- **Private (encrypted) bucket**: Contents are encrypted before upload and available only to authorized users.
+
+## Files: From Bytes to Commitments
+
+During upload, DataHaven runs a short pipeline as follows that turns raw bytes into a verifiable on-chain commitment:
+
+1. **Chunking:** The file is split deterministically (fixed-size or strategy-defined).
```

Comment:
Each chunk is 1KB afaik, maybe that's cool to add here.
Not sure if "strategy-defined" means chunk size can be customized - afaik it's 1KB and that's it